src/util/debounced.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * function that enforces programming practice used to ensure that
+ * time-consuming task do not fire so often, that it stalls the performance
+ * of the web page. Limiting the rate at which a function gets invoked.
```

Comment:
this is a very wordy way of saying "Ensures a function is only run only once every `delay`" 😅

Reply:
which is also a wordy way of saying "Debounces a function for `delay` ms"

Reply:
done